java/src/jni/h5rImp.c

```diff
@@ -204,6 +204,12 @@ Java_hdf_hdf5lib_H5_H5Rdestroy(JNIEnv *env, jclass clss, jbyteArray ref)
     if ((status = H5Rdestroy(&loc_ref)) < 0)
         H5_LIBRARY_ERROR(ENVONLY);
 
+    /* Reset buffer to 0 to be safe. H5Rdestroy does this, but
+     * only on our temporary reference object that we copied
+     * into for alignment reasons.
+     */
+    memset(refBuf, 0, H5R_REF_BUF_SIZE);
```

Comment:
@byrnHDF This PR may be related to the issue you've seen with references inside container types. After fixing the logic in nativeData.c to use pointers with the right alignment, I got a test failure in the Java region reference example from HDF5Examples due to a double free from `H5Rdestroy`.

This memset does what `H5Rdestroy` does above, except the call above was only doing it on the temporary reference variable that gets memcpy'd into, so we weren't clearing the passed in reference buffer after `H5Rdestroy`. Not a big deal as long as you don't try to call `H5Rdestroy` on the same buffer twice in Java, but the fact the example passed before tells me we were likely working with incorrectly-aligned pointers in this case.